### PR TITLE
Use depot

### PR DESCRIPTION
**Pull request by Sbozzolo**
Using a depot can reduce the init time to 3/4 minutes (from 25).

### CODE REVIEW

**Review by juliasloan25 — APPROVED** [review]

**Comment by Sbozzolo** [comment]
This PR also updates MPI to use MPITrampoline, allowing us to use a newer version of HDF5.

**Comment by Sbozzolo** [comment]
> This PR also updates MPI to use MPITrampoline, allowing us to use a newer version of HDF5.
> 
> (Following [CliMA/ClimaAtmos.jl#2377](https://github.com/CliMA/ClimaAtmos.jl/pull/2377))

I cannot update HDF5 because it is pinned by other dependendices, so this will have to go through after #535

File: .buildkite/longruns/pipeline.yml [hunk 1]
```diff
@@ -15,7 +15,11 @@ env:
   CONFIG_PATH: "config/longrun_configs"
   PERF_CONFIG_PATH: "config/perf_configs"
 
-  LD_LIBRARY_PATH: "/central/scratch/esm/slurm-buildkite/climaatmos-ci/depot/cpu/artifacts/e9feeabfa989549ac1dac61e4df5f2849149ad6d/lib/:${LD_LIBRARY_PATH}"
+  JULIA_NVTX_CALLBACKS: gc
```
Comment (juliasloan25) [inline .buildkite/longruns/pipeline.yml:18]: this line is duplicated above (line 10)

Comment (Sbozzolo) [reply .buildkite/longruns/pipeline.yml:17]: Fixed

**Pull request merged**